pyproject.toml

```diff
@@ -10,27 +10,42 @@ force-exclude = '''
 [tool.codespell]
 ignore-words-list = 'astroid,crasher,asend'
 
-[tool.flake8]
-extend-ignore = ['D', 'E', 'W', 'F403', 'F405', 'F821', 'F822']
-per-file-ignores = [
-  'trio/__init__.py: F401',
-  'trio/_core/__init__.py: F401',
-  'trio/_core/_tests/test_multierror_scripts/*: F401',
-  'trio/abc.py: F401',
-  'trio/lowlevel.py: F401',
-  'trio/socket.py: F401',
-  'trio/testing/__init__.py: F401'
+[tool.ruff]
+target-version = "py38"
+src = ["trio", "notes-to-self"]
+respect-gitignore = true
+fix = true
+
+select = [
+  "E",  # pycodestyle errors
+  "W",  # pycodestyle warnings
+  "F",  # pyflakes
+  "I",  # isort
+  "D",  # pydocstyle
+  "YTT",  # flake8-2020
 ]
+extend-ignore = ['D', 'E', 'W', 'F403', 'F405', 'F821', 'F822']
 
-[tool.isort]
-combine_as_imports = true
-profile = "black"
-skip_gitignore = true
-skip_glob = [
+include = ["*.py", "*.pyi", "**/pyproject.toml"]
+
+extend-exclude = [
   "docs/source/reference-*",
-  "docs/source/tutorial/*"
+  "docs/source/tutorial/*",
 ]
 
+[tool.ruff.per-file-ignores]
+'trio/__init__.py' = ['F401']
+'trio/_core/__init__.py' = ['F401']
+'trio/_core/_tests/test_multierror_scripts/*' = ['F401']
+'trio/abc.py' = ['F401']
+'trio/lowlevel.py' = ['F401']
+'trio/socket.py' = ['F401']
+'trio/testing/__init__.py' = ['F401']
+'trio/_core/_generated_*.py' = ['I']
```

Comment:
`_generated_*` shouldn't have `isort` ignored ... except you're gonna have to go modify `gen_exports` to use ruff instead of `isort`. See #2759 
That also means that `isort` can be removed from `test-requirements.in`, otherwise we're not gaining much anything.

Reply:
Using ruff instead of isort as of 06bcb005be777ff0aa6093e16909c30b115c7e64, still need to remove isort from test-requirements.in though.

Reply:
Looking through https://github.com/astral-sh/ruff/issues/3694, `ruff` supports the same sort of CLI API that `black` does - pass `-` to use stdin as the source, then get results on `stdout`.